### PR TITLE
#125 Adding pa11y-ci

### PR DESCRIPTION
**Pull request by adamdavies1**
# Description

Get A11y running on the pipeline using pa11y-ci

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce
the testing if necessary.

Before PR's can be merged they will need to be tested by QA and approved where
applicable. To flag the change to QA assign **@XGovFormBuilder/qa** as one of the reviewers.

- [ ] Ran the Branch QA work flow, there should be a new section in there called pa11y. You will see Accessibility errors on certain pages.

# Checklist:

- [x] My changes do not introduce any new linting errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and versioning
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have rebased onto main and there are no code conflicts
- [ ] I have checked deployments are working in all environments
- [ ] I have updated the architecture diagrams as per Contribute.md

### CODE REVIEW

**Comment by adamdavies1** [comment]
smoke test is failing for this branch due to the issues discovered by running pa11y. Perhaps we need to raise a new task to fix these?

**Comment by jenbutongit** [comment]
@adamdavies1 yep. If you're able to, would be good to fix them in this PR. if not, log them as issues mentioning this pull request. Looks like pa11y is working though - nice one!

**Comment by adamdavies1** [comment]
> @adamdavies1 yep. If you're able to, would be good to fix them in this PR. if not, log them as issues mentioning this pull request. Looks like pa11y is working though - nice one!

@jenbutongit should have a nice green pipeline now :)

File: designer/client/designer.tsx [hunk 1]
```diff
@@ -99,7 +99,7 @@ export default class Designer extends Component<Props, State> {
       <FeatureFlagProvider>
         <DataContext.Provider value={dataContextProviderValue}>
           <FlyoutContext.Provider value={flyoutContextProviderValue}>
-            <div id="app">
+            <div id="designer">
```
Comment (superafroman) [inline designer/client/designer.tsx:102]: `#app` has styles associated with it in `index.scss`, does dropping them have any impact?

File: designer/client/components/Menu/SubMenu.tsx [hunk 2]
```diff
@@ -72,7 +72,13 @@ export function SubMenu({ id, updateDownloadedAt }: Props) {
       >
         Download form
       </button>
-      <input ref={fileInput} type="file" hidden onChange={onFileUpload} />
+      <input
+        ref={fileInput}
+        type="file"
+        hidden
+        onChange={onFileUpload}
+        aria-label="country"
```
Comment (superafroman) [inline designer/client/components/Menu/SubMenu.tsx:80]: What's the significance of 'country'? This input accepts a form config to upload?

File: designer/client/designer.tsx (hunk 1 above)
Comment (adamdavies1) [reply designer/client/designer.tsx:102]: Doesn't seem to have any inpact. The issue was there was two divs similar to 
`<div id="app">
    <div id="app">`
We could duplicate the style for designer?

**Review by jenbutongit — CHANGES_REQUESTED** [review]
thanks @adamdavies1 - just needs that line removing. rest lgtm

File: designer/client/components/Menu/SubMenu.tsx (hunk 2 above)
Comment (adamdavies1) [reply designer/client/components/Menu/SubMenu.tsx]: aria-label="Import saved form" should now be seeing this

**Review by jenbutongit — APPROVED** [review]

**Pull request merged**